src/app/core/services/course/course.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ICourse } from '@features/courses/models/courses.model';
+import { MatDialog, MatDialogRef } from '@angular/material';
+import { DeleteDialogComponent } from '@features/dialogs/delete-dialog/delete-dialog.component';
+import { EditDialogComponent } from '@features/dialogs/edit-dialog/edit-dialog.component';
+import { Observable, Subject } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root'
```

Comment:
Course service will be using just in 'courses' module, please move this service to 'courses' and remove 'providedIn: 'root''. You could provide this one through features.module.ts (or create courses.module.ts in features/courses folder)